aspnetcore/performance/caching/memory.md

```diff
@@ -1,164 +1,106 @@
 
-Possible `eviction reasons` are:
+[!code-csharp[Main](memory/sample/WebCache/Controllers/HomeController.cs?name=snippet2&highlight=3-7,14-19)]
 
-**None**
+The following code calls [Get](https://docs.microsoft.com/en-us/aspnet/core/api/microsoft.extensions.caching.memory.cacheextensions#Microsoft_Extensions_Caching_Memory_CacheExtensions_Get__1_Microsoft_Extensions_Caching_Memory_IMemoryCache_System_Object_) to fetch the cached time:
 
-No reason known.
+[!code-csharp[Main](memory/sample/WebCache/Controllers/HomeController.cs?name=snippet_gct)]
 
-**Removed**
+See [IMemoryCache methods](https://docs.microsoft.com/en-us/aspnet/core/api/microsoft.extensions.caching.memory.imemorycache) and [CacheExtensions methods](https://docs.microsoft.com/en-us/aspnet/core/api/microsoft.extensions.caching.memory.cacheextensions) for a description of the cache methods.
 
-The item was manually removed by a call to `Remove()`
+## Using MemoryCacheEntryOptions
 
-**Replaced**
+The following sample:
 
-The item was overwritten.
+- Sets the absolute expiration time. This is the maximum time the entry can be cached and prevents the item from becoming too stale when the sliding expiration is continuously renewed.
+- Sets a sliding expiration time. Requests that access this cached item will reset the sliding expiration clock.
+- Sets the cache priority to `CacheItemPriority.NeverRemove`. 
+- Sets a [PostEvictionDelegate](https://docs.microsoft.com/en-us/aspnet/core/api/microsoft.extensions.caching.memory.postevictiondelegate) that will be called after the entry is evicted from the cache. The callback is run on a different thread from the code that removes the item from the cache.
 
-**Expired**
+[!code-csharp[Main](memory/sample/WebCache/Controllers/HomeController.cs?name=snippet_et&highlight=14-20)]
 
-The item timed out.
+## Cache dependencies
 
-**TokenExpired**
+The following sample shows how to expire a cache entry if a dependent entry expires. A `CancellationChangeToken` is added to the cached item. When `Cancel` is called on the `CancellationTokenSource`, both cache entries are evicted. 
 
-The token the item depended upon fired an event.
+[!code-csharp[Main](memory/sample/WebCache/Controllers/HomeController.cs?name=snippet_ed&highlight=11,25)]
 
-**Capacity**
+Using a `CancellationTokenSource` allows multiple cache entries to be evicted as a group. With the `using` pattern in the code above, cache entries created inside the `using` block will inherit triggers and expiration settings.
 
-The item was removed as part of the cache's memory management process.
+## Explicit cancellation with CancellationTokenSource
```

Comment:
@JunTaoLuo  OK, I'll remove it from the article. I'd like to keep it in the code as it's the only sample that shows how to explicitly cancel using the token. In the other sample using the cancellation token, the cache item expires and is evicted.